doc/developer/platform/ux.md

```diff
@@ -1,20 +1,945 @@
+
+dbt is essentially Terraform for databases. By supporting dbt, we eliminate much
+of the need for Terraform.
+
+However, while dbt has excellent support for managing views and indexes, and good
+support for managing sources and sinks, it is not the ideal tool for managing
+clusters, roles, databases, schemas, or secrets. So we might want to build a
+Terraform provider to manage these resources types.
+
+## APIs
+
+### Admin API
+
+The admin API is a REST API hosted at <https://admin.cloud.materialize.com>.
+It is provided by [Frontegg], our user management provider, and handles
+authentication and configuration of account security policies.
+
+The [Frontegg API reference][frontegg-api] specifies this API.
+
+The primary operation that this API provides is the exchange of user credentials
+for a short-lived authentication token. User credentials may be a username and
+password or an API key.
+
+We don't plan to publicize or document this API, but we don't plan to hide it,
+either.
+
+### Global API
+
+The global API is a REST API hosted at <https://cloud.materialize.com/api>. It
+handles operations that apply globally to an account rather than to a single
+region within an account.
+
+To authenticate with the global API, users supply an authentication token
+obtained from the admin API.
+
+We will make the following changes to the [existing API specification][global-api-spec]:
+
+  * Deprecate all operations on the `deployments` resource.
+  * Deprecate all operations on the `mz-versions` resource.
+  * Add a new API to enable a cloud-provider region for an account:
+
+    `PUT /api/region/{cloud-provider}/{region}`
+
+In the future, we will add billing operations to this API.
+
+As with the admin API, we don't plan to publicize or document the global API,
+but we don't plan to hide it, either.
+
+### Region API
+
+The user-facing API for Materialize Platform is the region API. This is the
+only API that we will actively advertise to users. The intent is that users
+think of the region API as **the** Materialize Platform API.
+
+**The region API is a SQL API.** It is not a REST API. While the API will
+be accessible via HTTPS, HTTPS is merely a container for a SQL payload.
+This is likely to be a contentious topic. Please see [Why SQL?](#why-sql)
+for justification.
+
+This implies that the web UI will communicate with the region API by using
+the SQL-over-HTTPS API.
+
+#### Endpoint
+
+The endpoint for the region API is a hostname of the following form:
+
+> CUSTOMER-ID.REGION.CLOUD-PROVIDER.materialize.cloud
+
+For example:
+
+> x32sk13nf.us-east-1.aws.materialize.cloud
+
+This endpoint requires TLS encryption. The server will present a TLS certificate
+issued by a trusted root CA.
+
+#### PostgreSQL server
+
+A PostgreSQL protocol server will be accessible at port 6875. The server
+processes SQL as described in [PostgreSQL Frontend/Backend Protocol][pgwire].
+
+The PostgreSQL server will support only username/password authentication. The
```

Comment:
Would supporting only username/password authentication prevent us from checking some checkbox in some future enterprise certification check list? What necessitates the removal of the cert-based certification as an option?

Reply:
It's actually kind of the opposite. Right now we have to manage our own TLS CA in order to issue client certificates to users, which is a fraught endeavor, and something that auditors look at with extreme suspicion. Switching to username/password authentication gets us out of the business of managing a TLS CA.

Down the road, if mTLS is a requirement for some customer's compliance, definitely not opposed to putting it back, but so far the deafening request has been "please god get rid of these TLS client certificates and give us a normal username and password."